### ISSUE TITLE
Write a model card for the fine-tuned bi-encoder (repository document, no hosted weights)

### ISSUE THREAD

**Issue by Dayallenr** [Dayallenr/RAG#25]
## Parent

#18 — Evaluate, publish, and serve the fine-tuned bi-encoder

## What to build

A model card as a **repository document**. Scope was changed in the comment below:
the fine-tuned weights stay local and gitignored, and nothing is published to a model
hub. What ships is the card itself, so a reader can judge the work without running
anything and without downloading anything.

The card is the artifact that turns a training run into a defensible claim. It documents
where the training data came from and how much of it there was, how the hard negatives
were mined, the split discipline that keeps validation honest (split by query, never by
row, because one query yields several triplets), the contamination guard, the
human-written test set held out entirely, the hardware, the hyperparameters, and the
measured delta.

The card states the delta whatever its sign, and states the confounds alongside it. The
relevance labels are a floor rather than an estimate, and the questions were drafted by
reading the passages they are labelled against — which shares vocabulary with them and
structurally favours lexical matching. A card quoting the number without those two
caveats overstates the result.

It also states what the no-hosted-checkpoint decision costs. Without published weights,
the fine-tune claim rests on the training script, the tracked training-split sample, and
`results/`. That is the same evidence standard as every other claim here, but it is
weaker than a downloadable checkpoint would have been, and the card says so rather than
leaving a reader to notice.

## Acceptance criteria

- [ ] The card is a document in this repository (e.g. `docs/model-card.md`), not a hub release — no weights are published
- [ ] The card documents the base model, the synthetic query generation method and volume, and the hard-negative mining strategy
- [ ] The card documents the split-by-query discipline, the contamination guard, and the held-out human test set
- [ ] The card documents the training hardware and hyperparameters, and says what ties them to the checkpoint the delta was measured with — `results/training/report.json` records no checkpoint identity, so unless `results/training/checkpoint.json` exists (written by `scripts/transfer_checkpoint.py push`) the card must state that the weights are verified as a fine-tune of the base model but are not cryptographically tied to that run
- [ ] The card states the measured delta with its sign, whatever that sign is, on the held-out test split from #32
- [ ] The card states the evaluation's known confounds, so the number reads as a floor rather than an estimate
- [ ] The card states the intended use and the known limitations
- [ ] The card states the consequence of shipping no weights: the claim rests on the training script, the tracked training-split sample, and `results/` — weaker than a downloadable checkpoint, partly compensated by the third-party-hosted tracking project in #27
- [ ] The project documentation links the card, and every claim in it maps to a real report

## Blocked by

- #23 — Measure the delta across the four-run matrix (the card must state a real, measured number)

**Comment by Dayallenr** [comment]
Scope changed: **no hosted checkpoint.** The fine-tuned weights stay local and gitignored, and nothing is published to a model hub.

What survives is the model card as a repository document — base model, training-data provenance, the contamination guard, the measured delta on the held-out test split from #32, known limitations, and intended use.

Stated consequence, so it is not discovered later: without published weights, the fine-tune claim rests on the training script, the tracked training-split sample, and `results/`. That is the same evidence standard as every other claim here, but it is weaker than a downloadable checkpoint would have been. The public experiment-tracking project linked in #27 partly compensates, since it is third-party hosted.

Retitle to reflect that this is a repository document rather than a release.

**Comment by Dayallenr** [comment]
Body reconciled with the scope-change comment above (2026-08-18). The two acceptance criteria that required publishing to a model hub and loading the published checkpoint are gone; the card is a repository document. Added criteria the comment implied but the body did not carry: the contamination guard, intended use and limitations, and an explicit statement of what shipping no weights costs the claim.

Also added a provenance criterion the comment predates: `results/training/report.json` records no checkpoint identity, so the card cannot honestly cite the training hyperparameters against the weights on the Mac unless `results/training/checkpoint.json` lands (`scripts/transfer_checkpoint.py push` on the 5070, then commit). Until it does, the card states the weights are verified as a fine-tune of the base model — 384-dim, normalized, moved from base at cos 0.75–0.86 — but are not tied to this run. That is a caveat to write, not a blocker: it does not gate #23.

**Comment by Dayallenr** [comment]
Done in e2714c7 — `docs/model-card.md`, linked from the README.

**All nine acceptance criteria are met.** Repository document, no weights published. It covers the base model, the 4,776 synthetic queries (`qwen3:8b` via local Ollama, stratified across company × chunk type), hard-negative mining (live hybrid retriever, candidate depth 20, top hit skipped, 3 negatives/query → 12,758 train + 1,429 val), split-by-query, the two-stage contamination guard, the fully held-out 101 human questions with their frozen 71/30 dev/test split, the hardware and hyperparameters, the delta with both signs, the confounds, intended use and limitations, and what shipping no weights costs.

**The provenance criterion changed while I was writing it, and two bugs were hiding behind that.**

`results/training/checkpoint.json` *is* in the repository — it landed in `44f4395`, after the ticket body and after CLAUDE.md's note saying it was absent. So the criterion's "unless the manifest exists" branch no longer applied and I had to actually run the check. It failed, for two different reasons:

1. **`transfer_checkpoint.py` keyed the manifest with the writing machine's path separator.** `str(relative)` on the Windows 5070 stored `1_Pooling\config.json`; read here, both nested module configs came back as *missing* **and** *unexpected*. `verify` could never pass for a manifest written on the training box — which is the only case the file exists for, and which is indistinguishable from weights that genuinely did not arrive. Keys are POSIX now, and `verify_against_manifest` re-normalises on the way in so the already-committed manifest still verifies.

2. **The delta report's `weights_traceable_to_this_run` was presence-only** — `bool(manifest and manifest["files"])`, never comparing a digest. Regenerating the report with the manifest now present would have flipped it to a confident, false *"these weights are tied to the training run that reported these losses."* It now takes the real `verify` output, treats only an empty problem list as a tie, and digests the model the fine-tuned arm actually named rather than a constant path. Shared digest logic moved to `duediligence/train/checkpoint.py` so the script and the report cannot drift.

**With the spurious noise gone, the real state:**

```
$ python scripts/transfer_checkpoint.py verify
models/bge-small-duediligence does not match results/training/checkpoint.json:
  digest mismatch: modules.json (expected 5861cc45c276…, got a0f5ee40975a…)
```

9 of 10 files match — including all **133,462,128 bytes of `model.safetensors`**, the tokenizer, the architecture config, and both nested module configs. **So the #23 delta is an attributable measurement of the trained model**, and the card cites the hyperparameters against these weights honestly.

The one mismatch is `modules.json`, 410 bytes, holding no weights and no hyperparameters. Two measured facts, not guesses: the checkpoint was written by sentence-transformers **6.0.0** (per `config_sentence_transformers.json`, which *does* match the manifest) while this Mac runs **5.7.0** and warns on load; and `modules.json` alone carries a distinct mtime (15:32 vs 22:29 for the files that verify), so it was rewritten locally *after* arrival, not in transit. The consistent account is a hand-edit into the 5.x format so it would load here — labelled an account, not a conclusion, because nobody recorded doing it.

`verify` still exits 1 and the card says so; this repo does not describe a failing check as a passing one. **Closing it means re-running `transfer_checkpoint.py manifest` on the 5070** against the checkpoint it still holds. Do **not** close it by re-digesting the local directory — that would make the manifest certify whatever is on this disk, which is the one thing it exists to prevent.

**Also corrected:** README and ADR 0005 both said the manifest "was never committed". Both now describe what `verify` actually reports.

**Verified:** `ruff check` clean; **556 tests pass** (was 543 — 4 cross-OS manifest tests, 2 traceability tests, 7 for the new checkpoint check). `results/finetune_delta/report.json` regenerated with `--from-reports` (no re-eval; the numbers reproduce identically: dense +0.233 test / +0.232 dev / +0.233 all, reranked +0.000). `scripts/verify_wandb_runs.py` re-run after that: **5,100/5,100 hosted metrics match**.

**Comment by Dayallenr** [comment]
**Follow-up: code review found five issues in the above, all fixed in `3ffe9e6`.** Two were real holes in the provenance check itself, and two were numbers this repo's own rules should have caught before I wrote them.

**1. `verify` could exit 0 having checked nothing.** Both loops in `verify_against_manifest` are driven by the manifest's file map, and `Path("/nope").rglob("*")` yields nothing rather than raising. So a manifest that lost its `files` map, checked against a directory the weights had not landed in, printed `verified 0 files` and **exited 0** — the single code this whole design uses to mean "these weights are the ones that were trained." An empty file map is now a problem in its own right.

**2. The delta report contradicted itself on malformed input.** `has_manifest` tested `manifest["files"]`; `_checkpoint_problems` guarded only on `manifest`. An empty file map therefore produced `checkpoint_manifest_present: false` next to a populated `checkpoint_problems` list and a note saying no manifest was present — three fields describing two different situations. Both now use the same test.

**3. The mtime argument was backwards.** I wrote that `modules.json` was "rewritten locally after the checkpoint arrived" because its mtime is distinct — 15:32 against 22:29 for the files that verify. But **15:32 is *older***, and older does not mean written later. The inference didn't follow from the evidence, and it sat under a heading promising measurement rather than inference.

The actual signal is better than the one I claimed: those 22:28–22:29 stamps are **ahead of this machine's own clock** (21:51 when checked), so they are timestamps preserved from elsewhere, while `modules.json` carries the one timestamp this machine could have produced. Same conclusion, now stated as consistent-with rather than proof, with an explicit note that mtimes cannot date the checkpoint's arrival at all. Corrected in the card, the README and ADR 0005.

**4. Two cosines were single samples quoted as measurements.** `results/index/report.json` samples 10 chunks; `cos_other_model` runs **0.486–0.884, mean 0.739**. `0.857` is one of them — quoted as *the* figure in the README (pre-existing) and propagated by me into the card and the claim-to-artifact map. All three now give the range.

**5. `cosine 0.68 against the base model` had no artifact.** I measured it in a shell this session, so it is not invented — but no report file produces it, and the card's own line 12 says every number comes from a report file named at the point of use. It was the one claim the document's own rule excluded. Dropped in favour of the `results/index/report.json` range, which supports the same point ("a real fine-tune, not a copy") from a real artifact.

**Verified:** `ruff check` clean, **563 tests pass** (was 556 — 4 for the vacuous-verify hole, 3 for the self-contradiction). Real-world behaviour is unchanged — `transfer_checkpoint.py verify` still exits 1 on `modules.json` alone, and `results/finetune_delta/report.json` still reports `present: true / traceable: false` with that one problem — so the fixes bite only on malformed input and no report needed regenerating.
